rollup/core/classes/RollupRepository.cls

```diff
@@ -90,80 +90,145 @@ public without sharing class RollupRepository implements RollupLogger.ToStringOb
+          DeveloperName,
+          LookupObject__c,
+          LookupObjectText__c,
+          LookupObject__r.QualifiedApiName,
+          CalcItem__c,
+          CalcItemText__c,
+          CalcItem__r.QualifiedApiName,
+          RollupFieldOnCalcItem__c,
+          RollupFieldOnCalcItemText__c,
+          RollupFieldOnCalcItem__r.QualifiedApiName,
+          LookupFieldOnCalcItem__c,
+          LookupFieldOnCalcItemText__c,
+          LookupFieldOnCalcItem__r.QualifiedApiName,
+          LookupFieldOnLookupObject__c,
+          LookupFieldOnLookupObjectText__c,
+          LookupFieldOnLookupObject__r.QualifiedApiName,
+          RollupFieldOnLookupObject__c,
+          RollupFieldOnLookupObjectText__c,
+          RollupFieldOnLookupObject__r.QualifiedApiName,
+          UltimateParentLookup__c,
+          UltimateParentLookup__r.QualifiedApiName,
+          CalcItemWhereClause__c,
+          ChangedFieldsOnCalcItem__c,
+          ConcatDelimiter__c,
+          CurrencyFieldMapping__c,
+          FullRecalculationDefaultNumberValue__c,
+          FullRecalculationDefaultStringValue__c,
+          GrandparentRelationshipFieldPath__c,
+          GroupByFields__c,
+          GroupByRowEndDelimiter__c,
+          GroupByRowStartDelimiter__c,
+          IsDisabled__c,
+          IsDistinct__c,
+          IsFullRecordSet__c,
+          IsRollupStartedFromParent__c,
+          IsTableFormatted__c,
+          LimitAmount__c,
+          OneToManyGrandparentFields__c,
+          OrderByFirstLast__c,
+          RollupControl__c,
+          RollupOperation__c,
+          RollupToUltimateParent__c,
+          SharingMode__c,
+          ShouldRunWithoutCustomSettingEnabled__c,
+          SplitConcatDelimiterOnCalcItem__c,
+          (SELECT Id, DeveloperName, FieldName__c, NullSortOrder__c, Ranking__c, SortOrder__c FROM RollupOrderBys__r),
+          RollupGrouping__r.Id,
+          RollupGrouping__r.RollupOperation__c
+        FROM Rollup__mdt
+        WHERE RollupControl__r.ShouldAbortRun__c = FALSE AND IsDisabled__c = FALSE
+        ORDER BY LookupObject__c
+      ];
+      // do the transforms for standard query
+      for (Rollup__mdt meta : matchingMetadata) {
+        meta.CalcItem__c = meta.CalcItem__r.QualifiedApiName ?? meta.CalcItemText__c;
+        meta.LookupFieldOnCalcItem__c = meta.LookupFieldOnCalcItem__r.QualifiedApiName ?? meta.LookupFieldOnCalcItemText__c;
+        meta.LookupFieldOnLookupObject__c = meta.LookupFieldOnLookupObject__r.QualifiedApiName ?? meta.LookupFieldOnLookupObjectText__c;
+        meta.RollupFieldOnCalcItem__c = meta.RollupFieldOnCalcItem__r.QualifiedApiName ?? meta.RollupFieldOnCalcItemText__c;
+        meta.RollupFieldOnLookupObject__c = meta.RollupFieldOnLookupObject__r.QualifiedApiName ?? meta.RollupFieldOnLookupObjectText__c;
+        meta.LookupObject__c = meta.LookupObject__r.QualifiedApiName ?? meta.LookupObjectText__c;
+        meta.UltimateParentLookup__c = meta.UltimateParentLookup__r.QualifiedApiName;
```

Comment:
can you update this to also use the safe navigator to fallback to the new `meta.UltimateParentLookupText__c`, following the same pattern as the others in this block? Apologies, I missed that addition when I originally reviewed from my phone.